var/spack/repos/builtin/packages/occa/package.py

```diff
@@ -50,13 +51,16 @@ class Occa(Package):
             description='Activates support for OpenCL')
     variant('coi',   default=False,
             description='Activates support for COI')
-    depends_on('cuda', when='+cuda')
-    
+    # depends_on('cuda', when='+cuda')
```

Comment:
Hi, I'm actually the one who wrote the CUDA package. It's pretty useful on Linux, but it looks like on macOS you're supposed to install CUDA via dmg, right? If we could add tty support to the install method (#2566), it would be pretty easy to die and tell the user that they need to install it manually.

I don't think the right solution here is to remove the CUDA dependency. I think we need to encourage users to add CUDA as an external package as @tgamblin mentioned. Once #2507 is finished, this will be even easier. We can print out a message on macOS that says "Install manually and run `spack external add ...`".

P.S. If we do move to arch-specific binaries, we'll have to address some of the limitations I brought up in #2009. If-statements can let you specify different versions for each operating system, but I think a `when=` parameter to the `version()` directive would be a better solution.

P.P.S. I know @svenevs was having a hell of a time installing CUDA on macOS due to the fact that NVIDIA doesn't support the latest compilers provided by Xcode on Sierra, so he may be able to give you pointers if you run into any problems.